p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
I've added another commit that eliminate the computeChecksum control block, and makes the headers have direction 'inout' in the Deparser control blocks, to enable header fields to be modified in those controls.